[poly2tri] Set policy CMP0063 to NEW

kduske opened the pull request:
**Describe the pull request**

- #### What does your PR fix?
Linker warnings about conflicting visibility settings if `poly2tri`, which is built as a static library, is pulled into a build as an indirect dependency of a shared library. An example of such a configuration is `assimp`, which depends on `poly2tri`, and can be built as a dynamic library on macOS.

Policy `CMP0063` enables CMake to set the symbol visibility of static libraries to avoid this problem.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  all, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes, to the best of my knowledge.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
 No, the maintainer guide says to do that only when updating multiple ports at the same time.

kduske committed:
Set policy CMP0063 to NEW

This change enables CMake to set the symbol visibility of static
libraries. Since poly2tri is compiled as a static library, without this
change, it can lead to linker warnings about conflicting visibility
settings if it is itself linked into a dynamic library.

kduske commented:
@microsoft-github-policy-service agree]

kduske commented:
@microsoft-github-policy-service agree

Neumann-A commented on ports/poly2tri/CMakeLists.txt:
```diff
@@ -1,5 +1,9 @@
 cmake_minimum_required(VERSION 3.0)
```
Just push this to 3.15 or higher. (vcpkg internally has 3.24)

kduske replied:
Do you mean instead of or in addition to setting the policy?

dg0yt replied:
> Do you mean instead of or in addition to setting the policy?

"instead". 
By requiring a particular version, CMake sets *all* policies introduced up to that version to `NEW`. Version 3.15 implicitly includes `CMP0063` from CMake 3.3.

kduske replied:
Unfortunately that does not fix the issue. As per [the documentation](https://cmake.org/cmake/help/latest/policy/CMP0063.html), this particular policy defaults to `OLD` if not set even on 3.25.

kduske replied:
I get the warning if I set

```
set(CMAKE_CXX_VISIBILITY_PRESET hidden)
set(CMAKE_VISIBILITY_INLINES_HIDDEN TRUE)
```

in `poly2tri`'s `CMakeList.txt`, but not the policy. And this reveals an error I made in the initial PR, where I did not set the visibility preset properties. I'm having a bit of trouble with my setup locally so I did not notice that it doesn't work when I only set the policy.

Here's the exact warning I get:

```
CMake Warning (dev) at /Users/kristian/Documents/Code/TrenchBroom/vcpkg/scripts/buildsystems/vcpkg.cmake:613 (_add_library):
  Policy CMP0063 is not set: Honor visibility properties for all target
  types.  Run "cmake --help-policy CMP0063" for policy details.  Use the
  cmake_policy command to set the policy and suppress this warning.

  Target "poly2tri" of type "STATIC_LIBRARY" has the following visibility
  properties set for CXX:

    CXX_VISIBILITY_PRESET
    VISIBILITY_INLINES_HIDDEN

  For compatibility CMake is not honoring them for this target.
Call Stack (most recent call first):
  CMakeLists.txt:38 (add_library)
This warning is for project developers.  Use -Wno-dev to suppress it.
```

kduske replied:
I'll push some fixup to address all these issues.

LilyWangLL commented on ports/poly2tri/vcpkg.json:
```diff
@@ -1,7 +1,7 @@
 {
   "name": "poly2tri",
   "version-string": "2020-07-21",
```
```suggestion
  "version-date": "2020-07-21",
```

kduske replied:
Thanks! I made the change in 06c684f6f02458249b458e885d0d50b264b5ce16.

kduske committed:
fixup! Set policy CMP0063 to NEW

kduske committed:
amend! Set policy CMP0063 to NEW

Set default symbol visibility to hidden

Policy CMP0063 enables CMake to set the symbol visibility of static
libraries. Since poly2tri is compiled as a static library, without this
change, the default symbol visibility cannot be set, leading to linker
warnings about conflicting visibility settings if poly2tri is itself
linked into a dynamic library.

BillyONeal commented:
Thanks for running down that extremely bizarre policy thing
